Occasional Embedded Social SDK crash after updating to target Android 8.0 (API level 26)

**Summary:** 

First mentioned in https://github.com/OneBusAway/onebusaway-android/pull/930#issuecomment-437492168.

After bumping to `targetSdkVersion 26`, when restarting the app from Android Studio I'm seeing this relatively frequently:

~~~
    Process: com.joulespersecond.seattlebusbot, PID: 21328
    java.lang.RuntimeException: Unable to create application org.onebusaway.android.app.Application: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.joulespersecond.seattlebusbot.BACKGROUND_INIT cmp=com.joulespersecond.seattlebusbot/com.microsoft.embeddedsocial.service.WorkerService (has extras) }: app is in background uid UidRecord{193ccd9 u0a700 CEM  idle procs:1 seq(0,0,0)}
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6059)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1764)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
     Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.joulespersecond.seattlebusbot.BACKGROUND_INIT cmp=com.joulespersecond.seattlebusbot/com.microsoft.embeddedsocial.service.WorkerService (has extras) }: app is in background uid UidRecord{193ccd9 u0a700 CEM  idle procs:1 seq(0,0,0)}
        at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1538)
        at android.app.ContextImpl.startService(ContextImpl.java:1484)
        at android.content.ContextWrapper.startService(ContextWrapper.java:663)
        at com.microsoft.embeddedsocial.base.service.ServiceLauncher.launchService(ServiceLauncher.java:49)
        at com.microsoft.embeddedsocial.base.service.ServiceLauncher.launchService(ServiceLauncher.java:39)
        at com.microsoft.embeddedsocial.sdk.EmbeddedSocial.init(EmbeddedSocial.java:102)
        at org.onebusaway.android.app.Application.setUpSocial(Application.java:581)
        at org.onebusaway.android.app.Application.onCreate(Application.java:111)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1125)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6056)
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1764) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6938) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
~~~

This crash is related to background executions limits introduced on Oreo:
https://developer.android.com/about/versions/oreo/background

**Steps to reproduce:** 

1. Use https://github.com/OneBusAway/onebusaway-android/pull/930 (or after that PR is merged, the master branch)
1. Launch the app from Android Studio
1. Make a code change, and re-launch the app from Android Studio while the app is still running

or

1. Restart your device
1. Long-press the OBA icon on your screen and move it around without dropping it

@acrown-msft is investigating to see if this occurs in other direct user-facing cases

**Expected behavior:** 

Not crash

**Observed behavior:** 

Crash

**Device and Android version:** 

Samsung Galaxy S8+ w/ Android 8.0

Here's another occurrence - if you long press the home screen icon, you'll get this stack trace:

~~~
2018-11-29 15:09:28.682 14940-15033/com.joulespersecond.seattlebusbot I/vndksupport: sphal namespace is not configured for this process. Loading /vendor/lib64/egl/libEGL_adreno.so from the current namespace instead.
2018-11-29 15:09:28.691 14940-15033/com.joulespersecond.seattlebusbot D/libEGL: loaded /vendor/lib64/egl/libEGL_adreno.so
2018-11-29 15:09:28.757 14940-14940/com.joulespersecond.seattlebusbot I/MultiDex: VM with version 2.1.0 has multidex support
2018-11-29 15:09:28.757 14940-14940/com.joulespersecond.seattlebusbot I/MultiDex: Installing application
2018-11-29 15:09:28.757 14940-14940/com.joulespersecond.seattlebusbot I/MultiDex: VM has multidex support, MultiDex support library is disabled.
2018-11-29 15:09:28.766 14940-15033/com.joulespersecond.seattlebusbot I/vndksupport: sphal namespace is not configured for this process. Loading /vendor/lib64/egl/libGLESv1_CM_adreno.so from the current namespace instead.
2018-11-29 15:09:28.769 14940-15033/com.joulespersecond.seattlebusbot D/libEGL: loaded /vendor/lib64/egl/libGLESv1_CM_adreno.so
2018-11-29 15:09:28.955 14940-14940/com.joulespersecond.seattlebusbot D/FirebaseApp: com.google.firebase.auth.FirebaseAuth is not linked. Skipping initialization.
2018-11-29 15:09:28.980 14940-14940/com.joulespersecond.seattlebusbot D/FirebaseApp: com.google.firebase.crash.FirebaseCrash is not linked. Skipping initialization.
2018-11-29 15:09:29.099 14940-14940/com.joulespersecond.seattlebusbot V/FA: Registered activity lifecycle callback
2018-11-29 15:09:29.112 14940-14940/com.joulespersecond.seattlebusbot I/FirebaseInitProvider: FirebaseApp initialization successful
2018-11-29 15:09:29.155 14940-14940/com.joulespersecond.seattlebusbot I/InstantRun: starting instant run server: is main process
2018-11-29 15:09:29.383 14940-14940/com.joulespersecond.seattlebusbot I/System.out: O311C: INFO Log level changed: DEBUG
2018-11-29 15:09:29.384 14940-14940/com.joulespersecond.seattlebusbot I/System.out: O311C: INFO Dry run mode changed: true
2018-11-29 15:09:29.384 14940-14940/com.joulespersecond.seattlebusbot W/Application: Open311 issue reporting is in debug/dry run mode - no issues will be submitted.
2018-11-29 15:09:29.384 14940-14940/com.joulespersecond.seattlebusbot I/System.out: O311C: INFO clearOpen311
2018-11-29 15:09:29.486 14940-15060/com.joulespersecond.seattlebusbot V/FA: Collection enabled
2018-11-29 15:09:29.486 14940-15060/com.joulespersecond.seattlebusbot V/FA: App package, google app id: com.joulespersecond.seattlebusbot, 1:297287630784:android:47d1e05cfc58f0b8
2018-11-29 15:09:29.488 14940-15060/com.joulespersecond.seattlebusbot I/FA: App measurement is starting up, version: 12451
2018-11-29 15:09:29.488 14940-15060/com.joulespersecond.seattlebusbot I/FA: To enable debug logging run: adb shell setprop log.tag.FA VERBOSE
2018-11-29 15:09:29.488 14940-15060/com.joulespersecond.seattlebusbot I/FA: To enable faster debug mode event logging run:
      adb shell setprop debug.firebase.analytics.app com.joulespersecond.seattlebusbot
2018-11-29 15:09:29.488 14940-15060/com.joulespersecond.seattlebusbot D/FA: Debug-level message logging enabled
2018-11-29 15:09:29.704 14940-15060/com.joulespersecond.seattlebusbot V/FA: Connecting to remote service
2018-11-29 15:09:29.850 14940-15060/com.joulespersecond.seattlebusbot V/FA: Connection attempt already in progress
2018-11-29 15:09:30.031 14940-15033/com.joulespersecond.seattlebusbot I/vndksupport: sphal namespace is not configured for this process. Loading /vendor/lib64/egl/libGLESv2_adreno.so from the current namespace instead.
2018-11-29 15:09:30.048 14940-15033/com.joulespersecond.seattlebusbot D/libEGL: loaded /vendor/lib64/egl/libGLESv2_adreno.so
2018-11-29 15:09:30.146 14940-15094/com.joulespersecond.seattlebusbot W/Settings: Setting airplane_mode_on has moved from android.provider.Settings.System to android.provider.Settings.Global, returning read-only value.
2018-11-29 15:09:30.397 14940-14940/com.joulespersecond.seattlebusbot D/AndroidRuntime: Shutting down VM
2018-11-29 15:09:30.403 14940-14940/com.joulespersecond.seattlebusbot E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.joulespersecond.seattlebusbot, PID: 14940
    java.lang.RuntimeException: Unable to create application org.onebusaway.android.app.Application: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.joulespersecond.seattlebusbot.BACKGROUND_INIT cmp=com.joulespersecond.seattlebusbot/com.microsoft.embeddedsocial.service.WorkerService (has extras) }: app is in background uid UidRecord{2c54c68 u0a735 CEM  idle procs:1 seq(0,0,0)}
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6059)
        at android.app.ActivityThread.-wrap1(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1764)
        at android.os.Handler.dispatchMessage(Handler.java:105)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6938)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374)
     Caused by: java.lang.IllegalStateException: Not allowed to start service Intent { act=com.joulespersecond.seattlebusbot.BACKGROUND_INIT cmp=com.joulespersecond.seattlebusbot/com.microsoft.embeddedsocial.service.WorkerService (has extras) }: app is in background uid UidRecord{2c54c68 u0a735 CEM  idle procs:1 seq(0,0,0)}
        at android.app.ContextImpl.startServiceCommon(ContextImpl.java:1538)
        at android.app.ContextImpl.startService(ContextImpl.java:1484)
        at android.content.ContextWrapper.startService(ContextWrapper.java:663)
        at com.microsoft.embeddedsocial.base.service.ServiceLauncher.launchService(ServiceLauncher.java:49)
        at com.microsoft.embeddedsocial.base.service.ServiceLauncher.launchService(ServiceLauncher.java:39)
        at com.microsoft.embeddedsocial.sdk.EmbeddedSocial.init(EmbeddedSocial.java:102)
        at org.onebusaway.android.app.Application.setUpSocial(Application.java:590)
        at org.onebusaway.android.app.Application.onCreate(Application.java:118)
        at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1125)
        at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6056)
        at android.app.ActivityThread.-wrap1(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1764) 
        at android.os.Handler.dispatchMessage(Handler.java:105) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6938) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:327) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1374) 
~~~

--- a/onebusaway-android/build.gradle
+++ b/onebusaway-android/build.gradle
@@ -449,8 +449,8 @@ dependencies {
     // Amazon Maps (only for Amazon flavor)
     amazonImplementation 'com.amazon.android:amazon-maps-api:2.0'
     // Embedded Social SDK
-    implementation('com.acrowntest.test:sdk:0.7.1:release@aar') {
-        transitive = true;
+    implementation('com.acrowntest.test:sdk:0.7.2:release@aar') {
+        transitive = true
     }
     // Required to support multidex for minSdkVersion <= 20
     implementation 'androidx.multidex:multidex:2.0.0'

--- a/onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java
+++ b/onebusaway-android/src/main/java/org/onebusaway/android/app/Application.java
@@ -587,7 +587,7 @@ public class Application extends MultiDexApplication {
     private void setUpSocial() {
         if (EmbeddedSocialUtils.isBuildVersionSupportedBySocial() &&
                 EmbeddedSocialUtils.isSocialApiKeyDefined()) {
-            EmbeddedSocial.init(this, R.raw.embedded_social_config, BuildConfig.EMBEDDED_SOCIAL_API_KEY);
+            EmbeddedSocial.init(this, R.raw.embedded_social_config, BuildConfig.EMBEDDED_SOCIAL_API_KEY, null);
             EmbeddedSocial.setReportHandler(new SocialReportHandler());
             EmbeddedSocial.setNavigationDrawerHandler(new SocialNavigationDrawerHandler());
             EmbeddedSocial.setAppProfile(new SocialAppProfile());

